tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -270,23 +269,24 @@ public function testFile(string $file): void
 		$failures = [];
 
 		foreach ($asserts as $args) {
-			$assertType = array_shift($args);
-			$file = array_shift($args);
+			if ($args[0] === 'type') {
+				$file = $args[1];
 
-			if ($assertType === 'type') {
```

Comment:
phpstan is not able to narrow the array-shape of the tagged union of $args based on `array_shift()` (which is equivalent to `$args[0]`).
rewrote the logic so PHPStan is able to understand it